lib/mongoid/extensions/big_decimal.rb

```diff
@@ -67,7 +77,31 @@ def demongoize(object)
         #
         # @since 3.0.7
         def mongoize(object)
-          object && object.numeric? ? object.to_s : nil
+          if object
+            if Mongoid.map_big_decimal_to_decimal128
+              mongoize_to_decimal128(object)
+            elsif object.numeric?
+              object.to_s
+            end
+          end
+        end
+
+        private
+
+        def demongoize_from_decimal128(object)
+          if Mongoid.map_big_decimal_to_decimal128
+            object.to_big_decimal
+          else
+            raise Mongoid::Errors::UnmappedBSONType.new(object)
```

Comment:
See my comment [here](https://github.com/mongodb/mongoid/pull/4363#issuecomment-267926616). There is a potential loss of precision if a Decimal128 is roundtripped (and automatically converted to a Ruby BigDecimal)